File: dagger2-scopes-and-subcomponents-part1/app/src/main/java/com/example/dagger2scopeandsubcomponentspart1/AppComponent.java
package com.example.dagger2scopeandsubcomponentspart1;

import dagger.Component;

@Component(modules = {AppModule.class})
public interface AppComponent {

    public Warrior getWarrior();

    void inject(WarriorApplication warriorApplication);
}


////example with applicationscope
//@ApplicationScope
//@Component(modules = {AppModule.class})
//public interface AppComponent {
//
//    public Warrior getWarrior();
//
//    void inject(WarriorApplication warriorApplication);
////    void inject(MainActivity mainActivity);
//}

File: dagger2-scopes-and-subcomponents/app/src/main/java/com/example/dagger2scopesandsubcomponents/AppComponent.kt
package com.example.dagger2scopesandsubcomponents

import dagger.Component

//@Component(modules = [AppModule::class])
//interface AppComponent {
//    fun getWarrior(): Warrior
//}

@ApplicationScope
@Component(modules = [AppModule::class])
interface AppComponent {
    fun getWarrior(): Warrior
}